core/src/main/java/org/opensearch/sql/expression/aggregation/CountAggregator.java

```diff
@@ -64,11 +66,32 @@ public String toString() {
    */
   protected static class CountState implements AggregationState {
     private int count;
+    private final Set<ExprValue> distinctValues = new HashSet<>();
 
     CountState() {
       this.count = 0;
     }
 
+    public void count(ExprValue value, Boolean distinct) {
+      if (distinct) {
+        if (!duplicated(value)) {
+          distinctValues.add(value);
+          count++;
+        }
+      } else {
+        count++;
+      }
+    }
+
+    private boolean duplicated(ExprValue value) {
+      for (ExprValue exprValue : distinctValues) {
+        if (value.compareTo(exprValue) == 0) {
```

Comment:
Wondering why this for loop is required since Set is in use? Is there difference between `equals` and `compareTo` of ExprValue?

Reply:
I was not aware of the hash process of the HashSet but thought the comparisons would be simply between different ExprValue instances. You reminded me of that it should be good to pass the loop, since the add and comparisons are actually on the top of hashes. Removed it. Thanks!